### PR TITLE
[SUPPORT] pos_fast_reconcile: unofficial perfs improvements

### PR DESCRIPTION
**Pull request by kebeclibre**
For POS Sessions over several hundreds orders, the closing can take an extremely
long time, mainly for 2 reasons:
- The reconciliation is O(n²), which means that reconciling the move lines of
  the payments with the move lines of the session gets longer fast
- The creation of move lines for each payment takes a long time

For the `master` branch, we are currently developping a batch create mechanism
in the ORM that should allow for the creation of a big amount of move lines
without intermediary recomputes (among other things).

The reconciliation algorithm has already been merged in master and is now
O(n log(n)), which should improve performance for these cases significantly.

This being a stable branch, we do not have the luxury of merging this. Instead,
Odoo Support provides this 'fix module' for those that are interested.

In a nutshell:
- Hardcoded fast creation of move lines for payments
    Note that this bit could potentially be incompatible with custom modules that
    modify the schema for the account_move_line table
- Fast Reconcile Algorithm
    Only applied to the reconciliation of the POS Sessions move lines and *not*
    applied to the rest of the accounting mechanisms.

Note that this module is provided outside of the standard Odoo code and is not
covered by any warranty. See the disclaimer in the manifest of this module
for more information.

courstesy of dbo-odoo and csn-odoo

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by eabquina** [comment]
will the pos_fast_reconcile module work on Odoo 12 and 13 too? Thanks.

**Comment by C3POdoo** [comment]
Dear @kebeclibre,

Thank you for your contribution but the version 11.0 is no longer supported.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense for the upper version, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._
